cmake/kokkoskernels_tpls.cmake

```diff
@@ -0,0 +1,416 @@
+    ENDIF()
+  ENDIF()
+
+ENDMACRO()
+
+MACRO(kokkoskernels_find_imported NAME)
+  CMAKE_PARSE_ARGUMENTS(TPL
+   "INTERFACE;ALLOW_SYSTEM_PATH_FALLBACK"
+   "HEADER;LIBRARY;IMPORTED_NAME"
+   "HEADERS;LIBRARIES;HEADER_PATHS;LIBRARY_PATHS"
+   ${ARGN})
+
+  IF(NOT TPL_MODULE_NAME)
+    SET(TPL_MODULE_NAME TPL${NAME})
+  ENDIF()
+
+  IF (TPL_ALLOW_SYSTEM_PATH_FALLBACK)
+    SET(ALLOW_PATH_FALLBACK_OPT ALLOW_SYSTEM_PATH_FALLBACK)
+  ELSE()
+    SET(ALLOW_PATH_FALLBACK_OPT)
+  ENDIF()
+
+  IF (NOT TPL_IMPORTED_NAME)
+    IF (TPL_INTERFACE)
+      SET(TPL_IMPORTED_NAME ${NAME})
+    ELSE()
+      SET(TPL_IMPORTED_NAME KokkosKernels::${NAME})
+    ENDIF()
+  ENDIF()
+
+  SET(${NAME}_INCLUDE_DIRS)
+  IF (TPL_HEADER)
+    KOKKOSKERNELS_FIND_HEADER(${NAME}_INCLUDE_DIRS ${TPL_HEADER} ${NAME} ${ALLOW_PATH_FALLBACK_OPT} PATHS ${TPL_HEADER_PATHS})
+  ENDIF()
+
+  FOREACH(HEADER ${TPL_HEADERS})
+    KOKKOSKERNELS_FIND_HEADER(HEADER_FIND_TEMP ${HEADER} ${NAME} ${ALLOW_PATH_FALLBACK_OPT} PATHS ${TPL_HEADER_PATHS})
+    IF(HEADER_FIND_TEMP)
+      LIST(APPEND ${NAME}_INCLUDE_DIRS ${HEADER_FIND_TEMP})
+    ENDIF()
+  ENDFOREACH()
+
+  SET(${NAME}_LIBRARY)
+  IF(TPL_LIBRARY)
+    KOKKOSKERNELS_FIND_LIBRARY(${NAME}_LIBRARY ${TPL_LIBRARY} ${NAME} ${ALLOW_PATH_FALLBACK_OPT} PATHS ${TPL_LIBRARY_PATHS})
+  ENDIF()
+
+  SET(${NAME}_FOUND_LIBRARIES)
+  FOREACH(LIB ${TPL_LIBRARIES})
+    #we want the actual name, not the name -lblas, etc
+    SET(LIB_CLEAN ${LIB})
+    STRING(FIND "${LIB}" "-l" PREFIX_IDX)
+    IF (PREFIX_IDX STREQUAL "0")
+      STRING(SUBSTRING ${LIB} 2 -1 LIB_CLEAN)
+    ENDIF()
+
+    KOKKOSKERNELS_FIND_LIBRARY(${LIB}_LOCATION ${LIB_CLEAN} ${NAME} ${ALLOW_PATH_FALLBACK_OPT} PATHS ${TPL_LIBRARY_PATHS})
+    IF(${LIB}_LOCATION)
+      LIST(APPEND ${NAME}_FOUND_LIBRARIES ${${LIB}_LOCATION})
+    ELSE()
+      SET(${NAME}_FOUND_LIBRARIES ${${LIB}_LOCATION})
+      BREAK()
+    ENDIF()
+  ENDFOREACH()
+
+  INCLUDE(FindPackageHandleStandardArgs)
+  SET(TPL_VARS_NEEDED)
+  IF (TPL_LIBRARY)
+    LIST(APPEND TPL_VARS_NEEDED ${NAME}_LIBRARY)
+  ENDIF()
+  IF(TPL_HEADER)
+    LIST(APPEND TPL_VARS_NEEDED ${NAME}_INCLUDE_DIRS)
+  ENDIF()
+  IF(TPL_LIBRARIES)
+    LIST(APPEND TPL_VARS_NEEDED ${NAME}_FOUND_LIBRARIES)
+  ENDIF()
+  FIND_PACKAGE_HANDLE_STANDARD_ARGS(${TPL_MODULE_NAME} REQUIRED_VARS ${TPL_VARS_NEEDED})
+
+  IF (${TPL_MODULE_NAME}_FOUND)
+    SET(IMPORT_TYPE)
+    IF (TPL_INTERFACE)
+      SET(IMPORT_TYPE "INTERFACE")
+    ENDIF()
+    KOKKOSKERNELS_CREATE_IMPORTED_TPL(${TPL_IMPORTED_NAME}
+      ${IMPORT_TYPE}
+      INCLUDES "${${NAME}_INCLUDE_DIRS}"
+      LIBRARY  "${${NAME}_LIBRARY}"
+      LINK_LIBRARIES "${${NAME}_FOUND_LIBRARIES}")
+  ENDIF()
+ENDMACRO()
+
+MACRO(kokkoskernels_export_imported_tpl NAME)
+  IF (NOT KOKKOSKERNELS_HAS_TRILINOS)
+    GET_TARGET_PROPERTY(LIB_TYPE ${NAME} TYPE)
+    IF (${LIB_TYPE} STREQUAL "INTERFACE_LIBRARY")
+      # This is not an imported target
+      # This an interface library that we created
+      INSTALL(
```

Comment:
@jjwilke I'm running into issues during configuration when hitting this INSTALL call with a Cuda build due to CUBLAS.
By default, CUBLAS is enabled when Cuda is enabled (I think this was preexisting behavior).
When this install call is hit I see errors like:

```
CMake Error at cmake/kokkoskernels_tpls.cmake:273 (INSTALL):
  INSTALL TARGETS given target "KokkosKernels::CUBLAS" which is an alias.
Call Stack (most recent call first):
  cmake/kokkoskernels_tpls.cmake:352 (KOKKOSKERNELS_EXPORT_IMPORTED_TPL)
  cmake/kokkoskernels_tpls.cmake:437 (KOKKOSKERNELS_IMPORT_TPL)
  CMakeLists.txt:371 (INCLUDE)
```

This is for `TARGETS` set to `KokkosKernels::CUBLAS`
The `${CMAKE_INSTALL_BINDIR}` and `${CMAKE_INSTALL_LIBDIR}` are empty strings, though in the FindTPLCUBLAS module `${CUDA_CUBLAS_LIBRARIES}` returns the path to the cublas shared lib.

Reply:
I pushed a fix. I will make this more robust in 3.1. Stupid mistake on my part.

Reply:
Just tested to confirm it is working, thanks for fast fix!